homeassistant/components/mqtt/debug_info.py

```diff
@@ -21,7 +21,7 @@ def log_messages(hass: HomeAssistantType, entity_id: str) -> MessageCallbackType
     def _log_message(msg):
         """Log message."""
         debug_info = hass.data[DATA_MQTT_DEBUG_INFO]
-        messages = debug_info["entities"][entity_id]["topics"][msg.topic]
+        messages = debug_info["entities"][entity_id]["topics"][msg.subscribed_topic]
```

Comment:
Isn't this weird? I would expect to know the topic the message was received on, not perse the subscription ?